pkg/utils/convert_test.go

```diff
@@ -0,0 +1,115 @@
+package utils
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConvertToFloat64(t *testing.T) {
+	type tt struct {
+		input   interface{}
+		wantf64 float64
+		wantu32 uint32
+		wantu64 uint64
+		wanti64 int64
+		wanti   int
+	}
+	cases := []tt{{
+		input:   float64(1.1),
+		wantf64: 1.1,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   float32(1.1),
+		wantf64: 1.1,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   "1",
+		wantf64: 1,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   int32(1),
+		wantf64: 1.0,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   int64(1),
+		wantf64: 1.0,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   int(1),
+		wantf64: 1.0,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   uint32(1),
+		wantf64: 1.0,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   uint64(1),
+		wantf64: 1.0,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   uint(1),
+		wantf64: 1.0,
+		wantu32: 1,
+		wantu64: 1,
+		wanti64: 1,
+		wanti:   1,
+	}, {
+		input:   time.Duration(42),
+		wantf64: 42.0,
+		wantu32: 42,
+		wantu64: 42,
+		wanti64: 42,
+		wanti:   42,
+	}}
+	for _, tc := range cases {
+		tc := tc
+		t.Run(fmt.Sprintf("%t", tc.input), func(t *testing.T) {
```

Comment:
`%t` prints the word true or false. Perhaps `%T` was meant here? (`%T` prints a Go-syntax representation of the type of the value)
Reference:
https://pkg.go.dev/fmt#hdr-Printing

Reply:
oops good catch ... for some reason it wasn't writing true/false, but the some garbage AND the type so I didn't pay much attention